src/layers/models/layersManager.ts

```diff
@@ -35,29 +37,65 @@ class LayersManager {
     const jsonDocument: IMapProxyJsonDocument = await this.configProvider.getJson();
 
     if (!isLayerNameExists(jsonDocument, layerName)) {
-      throw new NotFoundError(`Layer name '${layerName}' is not exists`);
+      throw new NotFoundError(`Layer name '${layerName}' does not exist`);
     }
     const requestedLayer: IMapProxyCache = jsonDocument.caches[layerName] as IMapProxyCache;
     return requestedLayer;
   }
 
   public async addLayer(layerRequest: ILayerPostRequest): Promise<void> {
-    this.logger.info(`Add layer request: ${layerRequest.name}`);
-
     const editJson = (jsonDocument: IMapProxyJsonDocument): IMapProxyJsonDocument => {
-      if (isLayerNameExists(jsonDocument, layerRequest.name)) {
-        throw new ConflictError(`Layer name '${layerRequest.name}' is already exists`);
-      }
-      const tileFormat = this.mapToTileFormat(layerRequest.format);
-      const newCache: IMapProxyCache = this.getCacheValues(layerRequest.cacheType, layerRequest.tilesPath, tileFormat);
-      const newLayer: IMapProxyLayer = this.getLayerValues(layerRequest.name);
-      jsonDocument.caches[layerRequest.name] = newCache;
-      jsonDocument.layers.push(newLayer);
+      this.addNewCache(jsonDocument, layerRequest);
+
       return jsonDocument;
     };
 
     await this.configProvider.updateJson(editJson);
-    this.logger.info(`Successfully added layer: ${layerRequest.name}`);
+  }
+
+  private addNewCache(jsonDocument: IMapProxyJsonDocument, layerRequest: ILayerPostRequest) {
+    const isRedisEnabled = config.get<boolean>('redis.enabled');
```

Comment:
get this in contructor - no need to resolve this every time this method is called